### PR TITLE
kubelet - node labels command line option removal

### PR DESCRIPTION
**Pull request by gambol99**
- removing the node-label command line option as requested in https://github.com/kubernetes/kubernetes/pull/17265#issuecomment-163045447

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1052,7 +1006,7 @@ func (kl *Kubelet) retrieveNodeLabelsFile(path string) (map[string]string, error
 
 	err = yaml.NewYAMLOrJSONDecoder(bufio.NewReader(fd), 12).Decode(&kps)
```
Comment (vishh) [inline pkg/kubelet/kubelet.go]: Where is the format for this file defined? Is it versioned? How can we introduce changes to this format?

**Comment by gambol99** [comment]
Hi @vishh, the format is indicated in the command line option, though yes a side note or doc would have been useful, versioned? given the what it does, apply's a one-time on registration only, series of labels i'm not sure I see the requirement for metadata such as versions. In regard to making changes I assumed this was looked at in the design / proposal highlighted in previous PR and rejected for the reasons you mentioned ... Either way, I appreciate your priorities and perspective is probably more holistic on this than mine ... Not wishing to push it any further, i'm happy to drop this PR and revert the previous one #17265

**Comment by vishh** [comment]
@gambol99: One approach to solve the versioning issue with label file is to create a new `KubeletConfiguration` object [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/v1alpha1/types.go) and use that as the label file format.
```go
type KubeletConfiguration struct {
  unversioned.TypeMeta
  NodeLabels map[string]string `json:"labels,omitempty"`
}
```
AFAIK, this format will be used to specify kubelet configuration in the future instead of command line flags. Will you be able to work on this?

**Comment by vishh** [comment]
cc @pwittrock

**Comment by pwittrock** [comment]
After some more discussion with vishh, using a KubeletConfiguration seems like the best long term strategy for configuring the labels, and the cli flag option is a better fit for this than the file.
I put together #19498 to remove the file and change the cli flag format.  This will make it possible for @mikedanese to include this in the KubeletConfiguration PR

**Pull request closed without merging**